libs/components/modals/src/lib/modules/modal/modal-host.component.ts

```diff
@@ -139,11 +155,38 @@ export class SkyModalHostComponent implements OnDestroy {
       injector
     );
 
+    // modal element that was just opened
+    const modalElement = modalComponentRef.location.nativeElement;
+
     modalInstance.componentInstance = modalComponentRef.instance;
 
     this.#registerModalInstance(modalInstance);
+    // hidding all elements at the modal-host level from screenreaders
+    const hostElement = this.#elRef.nativeElement;
+    let allModals = hostElement.children;
+
+    if (SkyModalHostService.openModalCount == 1) {
+      this.hideModalHostSiblings();
+      ariaPreviousValueSiblings = this.#ariaPreviousValueSiblings;
+    }
+    if (
+      SkyModalHostService.openModalCount > 1 &&
+      allModals[allModals.length - 1] == modalElement
+    ) {
+      allModals[allModals.length - 2].setAttribute('aria-hidden', true);
+    }
```

Comment:
I would use either `nextElementSibling` or `previousElementSibling` for this. https://developer.mozilla.org/en-US/docs/Web/API/Element/previousElementSibling